source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/teamexplorer/internal/TeamExplorerHelpers.java

```diff
@@ -279,4 +311,219 @@ public static IViewPart showView(final String viewID) {
+
+        if (n < 2 || !GIT_TOKEN.equals(pathItems[n - 2])) {
+            return false;
+        }
+
+        final String repository = pathItems[n - 1];
+        String path;
+        String project;
+
+        // Note that because uri.getPath() is absolute, pathItems[0] is always
+        // an empty string.
+        if (ServerURIUtils.isHosted(uri)) {
+            /*
+             * @formatter off            
+             * Possible URLs are:
+             * https://account.visualstudio.com/_git/repository                           (n=3)
+             * https://account.visualstudio.com/project/_git/repository                   (n=4) 
+             * https://account.visualstudio.com/DefaultCollection/_git/repository         (n=4) 
+             * https://account.visualstudio.com/DefaultCollection/project/_git/repository (n=5) 
+             * in the future, "DefaultCollection" will be replaced with "Organization"
+             * @formatter:on          
+             */
+            switch (n) {
+                case 5:
+                    project = pathItems[2];
+                    path = '/' + pathItems[1];
+                    break;
+                case 4:
+                    if (DEFAULT_COLLECTION_TOKEN.equalsIgnoreCase(pathItems[1])) {
+                        project = repository;
+                        path = '/' + pathItems[1];
+                    } else {
+                        project = pathItems[1];
+                        path = null;
+                    }
+                    break;
+                case 3:
+                    project = repository;
+                    path = null;
+                    break;
+                default:
+                    return false;
+            }
+        } else {
+            /*
+             * @formatter off            
+             * Possible URLs are:
+             * https://server:port/path/collection/_git/repository                           (n=5)
+             * https://server:port/path/collection/project/_git/repository                   (n=6) 
+             * @formatter:on          
+             */
+            switch (n) {
+                case 6:
+                    project = pathItems[3];
+                    path = '/' + pathItems[1] + '/' + pathItems[2];
+                    break;
+                case 5:
+                    project = repository;
+                    path = '/' + pathItems[1] + '/' + pathItems[2];
+                    break;
+                default:
+                    return false;
+            }
+        }
+
+        System.setProperty(
+            PROTOCOL_HANDLER_SERVER_PROPERTY,
+            URIUtils.newURI(uri.getScheme(), uri.getAuthority(), path, null, null).toString());
+        System.setProperty(PROTOCOL_HANDLER_REPOSITORY_PROPERTY, repository);
+        System.setProperty(PROTOCOL_HANDLER_PROJECT_PROPERTY, project);
+
+        return true;
+    }
+
+    public static void removeProtocolHandlerArguments() {
```

Comment:
Currently, I do not see any better place to keep this information. Created a task to revisit this later.